osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/infra/jms/messageprocessors/DomainResponseMessageProcessor.java

```diff
@@ -0,0 +1,128 @@
+        LOGGER.debug("Processing distribution automation response message");
+
+        String correlationUid = null;
+        String messageType = null;
+        String organisationIdentification = null;
+        String deviceIdentification = null;
+
+        ResponseMessageResultType resultType;
+        String resultDescription;
+        ResponseMessage dataObject;
+
+        try {
+            correlationUid = message.getJMSCorrelationID();
+            organisationIdentification = message.getStringProperty(Constants.ORGANISATION_IDENTIFICATION);
+            deviceIdentification = message.getStringProperty(Constants.DEVICE_IDENTIFICATION);
+
+            messageType = message.getJMSType();
+
+            resultType = ResponseMessageResultType.valueOf(message.getStringProperty(Constants.RESULT));
+            resultDescription = message.getStringProperty(Constants.DESCRIPTION);
+
+            dataObject = (ResponseMessage) message.getObject();
+        } catch (final IllegalArgumentException e) {
+            LOGGER.error("UNRECOVERABLE ERROR, received messageType {} is unknown.", messageType, e);
+            logDebugInformation(messageType, correlationUid, organisationIdentification, deviceIdentification);
+
+            return;
+        } catch (final JMSException e) {
+            LOGGER.error("UNRECOVERABLE ERROR, unable to read ObjectMessage instance, giving up.", e);
+            logDebugInformation(messageType, correlationUid, organisationIdentification, deviceIdentification);
+
+            return;
+        }
+
+        try {
+            LOGGER.info("Calling application service function to handle response: {}", messageType);
+
+            final CorrelationIds ids = new CorrelationIds(organisationIdentification, deviceIdentification,
+                    correlationUid);
+            this.handleMessage(ids, messageType, resultType, resultDescription, dataObject);
+
+        } catch (final RuntimeException e) {
+            handleError(e, correlationUid);
+        }
+    }
+
+    private void handleMessage(final CorrelationIds ids, final String messageType,
+            final ResponseMessageResultType resultType, final String resultDescription, final ResponseMessage message) {
+
+        final Serializable dataObject = message.getDataObject();
+        if (!(dataObject instanceof MeasurementReport)) {
+            LOGGER.error("For this component we only handle measurement reports");
+        } else {
+            this.meterReadingProducer.send(this.mapper.map(dataObject,
+                    org.opensmartgridplatform.domain.da.measurements.MeasurementReport.class));
+        }
+
+    }
+
+    /**
+     * In case of an error, this function can be used to send a response
+     * containing the exception to the web-service-adapter.
+     *
+     * @param e
+     *            The exception.
+     * @param correlationUid
+     *            The correlation UID.
+     * @param notificationType
+     *            The message type.
```

Comment:
there is no param notification Type

Reply:
No use of having this javadocs above a private method. Removed.